mix.exs

```diff
@@ -14,25 +15,46 @@ defmodule Stripe.Mixfile do
         "coveralls.post": :test,
         "coveralls.html": :test
       ],
-      deps: deps]
+      deps: deps
+    ]
   end
 
   # Configuration for the OTP application
   def application do
     [
-      applications: [:httpoison]
+      mod: {Stripe, []},
+      env: env(),
+      applications: apps(Mix.env)
     ]
   end
 
+  defp env() do
+    [
+      stripity_stripe: [
+        api_base_url: "https://api.stripe.com/v1/",
+        use_connection_pool: true,
+        pool_options: [
+          timeout: 5_000,
+          max_connections: 10
+        ]
+      ]
+    ]
+  end
+
+  defp apps(:test), do: [:bypass | apps()]
+  defp apps(_), do: apps()
+
+  defp apps(), do: [:hackney, :poison, :logger]
+
   defp deps do
     [
-      {:httpoison, ">= 0.0.0" },
-      {:poison, ">= 0.0.0", optional: true},
-      {:ex_doc, ">= 0.0.0", only: :dev},
-      {:earmark, ">= 0.0.0", only: :dev},
-      {:excoveralls, ">= 0.0.0", only: :test},
-      {:mock, ">= 0.0.0", only: :test},
-      {:inch_ex, ">= 0.0.0", only: [:dev, :test]}
+      {:bypass, "~> 0.5", only: :test},
+      {:earmark, "~> 1.0", only: :dev},
+      {:ex_doc, "~> 0.14", only: :dev},
+      {:excoveralls, "~> 0.5", only: :test},
+      {:hackney, "~> 1.6"},
+      {:inch_ex, "~> 0.5", only: [:dev, :test]},
+      {:poison, "~> 2.0"}
```

Comment:
I'm not worried about 1.5, but yeah... `"~> 2.0 or ~> 3.0"` will means it supports both 2.0 and 3.0